scRNA-seq-advanced/01-reading_filtering_CellRanger.Rmd

```diff
@@ -0,0 +1,520 @@
+
+```{r miQC plotMetrics}
+# use miQC::plotMetrics()
+miQC::plotMetrics(filtered_sce) + theme_bw()
+```
+
+We can see that there is a range of mitochondrial percentages, and it does also seem that cells with high percentages of mitochondrial genes don't seem to contain very many unique genes.
+
+How do we filter with this information?
+One option is to define a cutoff for the mitochondrial percentage above which we call a cell compromised and exclude it from further analysis. 
+However, choosing that cutoff can be a bit fraught, as the expected percentage of mitochondrial reads can vary depending on the cell type and library preparation methods. 
+So it might be nice to have a method to determine that cutoff from the data itself.
+
+### Filtering compromised cells
+
+Determining mitochondrial cutoffs is exactly what the `miQC` package does ([Hippen _et al._ 2021](https://doi.org/10.1371/journal.pcbi.1009290))!
+In truth, it does something possibly even a bit better: it fits a mixture model to the data that consists of distributions of healthy cells and compromised cells.
+Then we can calculate whether each cell is more likely to belong to the healthy or compromised distribution.
+We can then exclude the cells that are more likely to be compromised. 
+
+To use `miQC`, we first fit a model to the data in our SCE object:
+
+```{r miQC model}
+miqc_model <- miQC::mixtureModel(filtered_sce) 
+```
+
+Now we can plot the model results to see how it corresponds to our data.
+We should expect to see two fit lines: 
+
+- One line will correspond the the "healthy" cells and should show little to no relationship between the number of unique genes detected and the mitochondrial percentage.
+- By contrast, the line that corresponds to "compromised" cells will show a negative relationship between the number of unique genes detected and the mitochondrial percentage.
+
+This plot will also show, for each cell, the posterior probability that the cell is derived from the compromised distribution; a higher score indicates that a cell is more likely to be compromised.
+
+It is also critical to note that this model can _and does_ fail at times. 
+Plotting the results as we have done here is not a step to skip. 
+**Always look at your data!**
+
+```{r miQC plotModel}
+miQC::plotModel(filtered_sce, miqc_model) + theme_bw()
+```
+
+We can now filter our data based on the probability compromised as calculated from the model.
+But before we do that, we might want to quickly plot to see what would be filtered out with a given cutoff.
+The default is to exclude cells that have a posterior probability of 0.75 or greater of being compromised.
+
+
+```{r miQC plotFiltering}
+miQC::plotFiltering(filtered_sce, miqc_model, posterior_cutoff = 0.75)
```

Comment:
I might rephrase and/or add phrasing like - "even though 0.75 is the default let's write it to be explicit." Not critical.